cpp/arcticdb/version/version_core.cpp

```diff
@@ -669,15 +669,54 @@ std::shared_ptr<std::unordered_set<std::string>> columns_to_decode(const std::sh
     return res;
 }
 
-std::vector<RangesAndKey> generate_ranges_and_keys(const std::vector<SliceAndKey>& slice_and_keys) {
+std::vector<RangesAndKey> generate_ranges_and_keys(PipelineContext* const pipeline_context) {
```

Comment:
We don't generally use pointers unless the result can be null, is it possible to use a reference? A reference sort of implies that the lifetime is managed by the enclosing functions, whereas a pointer implies that the memory is entirely elsewhere (and might  be null or invalid)

The one place that we break the rule is with classes, where sometimes reference initialization of members can be annoying for PODs, as they are no longer trivially copyable etc